src/Microsoft.AspNetCore.SignalR.Redis/RedisOptions.cs

```diff
@@ -16,10 +17,10 @@ public class RedisOptions
             AbortOnConnectFail = false
         };
 
-        public Func<TextWriter, IConnectionMultiplexer> Factory { get; set; }
+        public Func<TextWriter, Task<IConnectionMultiplexer>> Factory { get; set; }
 
         // TODO: Async
-        internal IConnectionMultiplexer Connect(TextWriter log)
+        internal async Task<IConnectionMultiplexer> Connect(TextWriter log)
         {
             if (Factory == null)
```

Comment:
Assign factory to a local variable inside this method so there is no way the field can be set to null out from under you inbetween this test and using the factory.